x-pack/filebeat/input/azureblobstorage/mock/mock.go

```diff
@@ -0,0 +1,46 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package mock
+
+import (
+	"net/http"
+	"strings"
+)
+
+//nolint:errcheck // We can ignore as this is just for testing
```

Comment:
I'd prefer that this based the claim on the non-handleablility of response writer errors rather than it being testing.